types/database/channel.d.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Query, Relation} from '@nozbe/watermelondb';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+
+import ChannelInfo from '@typings/database/channel_info';
+import ChannelMembership from '@typings/database/channel_membership';
+import Draft from '@typings/database/draft';
+import GroupsInChannel from '@typings/database/groups_in_channel';
+import MyChannel from '@typings/database/my_channel';
+import MyChannelSettings from '@typings/database/my_channel_settings';
+import Post from '@typings/database/post';
+import PostsInChannel from '@typings/database/posts_in_channel';
+import Team from '@typings/database/team';
+import User from '@typings/database/user';
+
+/**
+ * The Channel model represents a channel in the Mattermost app.
+ */
+export default class Channel extends Model {
+    /** table (entity name) : Channel */
+    static table: string;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations;
+
+    /** create_at : The creation date for this channel */
+    createAt: number;
+
+    /** creator_id : The user who created this channel */
+    creatorId: string;
+
+    /** delete_at : The deletion/archived date of this channel */
+    deleteAt: number;
+
+    /** display_name : The channel display name (e.g. Town Square ) */
+    displayName: string;
+
+    /** is_group_constrained : If group is restricted to certain users/teams only */
+    isGroupConstrained: boolean;
+
+    /** name : The name of the channel (e.g town-square) */
+    name: string;
+
+    /** team_id : The team to which this channel belongs.  It can be empty for direct/group message. */
+    teamId: string;
+
+    /** type : The type of the channel ( e.g. G: group messages, D: direct messages, P: private channel and O: public channel) */
+    type: string;
+
+    /** members : Users belonging to this channel */
+    members: ChannelMembership[];
+
+    /** draft : All drafts for this channel */
+    drafts: Draft[];
+
+    /** groupsInChannel : Every group contained in this channel */
+    groupsInChannel: GroupsInChannel[];
+
+    /** posts : all posts made in that channel */
+    posts: Post[];
+
+    /** postsInChannel : a section of the posts for that channel bounded by a range */
+    postsInChannel: PostsInChannel[];
+
+    /** team : The TEAM to which this CHANNEL belongs */
+    team: Relation<Team>;
+
+    /** creator : The USER who created this CHANNEL*/
```

Comment:
if we haven't fetch the user from the server, this could also be null or undefined, right? unless we fetch this data for every channel as we get the channels from the server.

Reply:
We haven't defined optional fields in the schema. If a field is null/undefined, its value would be an empty string as in the case for the team_id.

To retrieve this value; we can : 

1. Fetch the value immediately - as you suggested.
2. Or fetch it lazily - only when needed -  and then do an update operation.

Each has its merits. Perhaps we can look into when we start with the business logic?